src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncEHTests.cs

```diff
@@ -2104,5 +2104,43 @@ static void Main()
 ";
             CompileAndVerify(source, expected);
         }
+
+        [Fact, WorkItem(67091, "https://github.com/dotnet/roslyn/issues/67091")]
+        public void NestedCatch_DuplicateWhen()
```

Comment:
>DuplicateWhen() [](http://example.com/codeflow?start=32&length=15)

Is it the duplicate `when` that causes the bug or is it the duplicated local name and type of the hoisted locals? (Just trying to confirm my understanding.)

Reply:
It's the duplicated local name and type. I just didn't know that yet when writing the test name. But having two `when` clauses is also needed, otherwise we couldn't have duplicate locals. I guess I will update the test name to make it clearer.
```suggestion
        public void NestedCatch_DuplicateLocal()
```